http/org_service.go

```diff
@@ -203,399 +201,232 @@ func newOrgResponse(o *influxdb.Organization) *orgResponse {
 			"tasks":      fmt.Sprintf("/api/v2/tasks?org=%s", o.Name),
 			"dashboards": fmt.Sprintf("/api/v2/dashboards?org=%s", o.Name),
 		},
-		Organization: *o,
-	}
-}
-
-type secretsResponse struct {
-	Links   map[string]string `json:"links"`
-	Secrets []string          `json:"secrets"`
-}
-
-func newSecretsResponse(orgID influxdb.ID, ks []string) *secretsResponse {
-	return &secretsResponse{
-		Links: map[string]string{
-			"org":  fmt.Sprintf("/api/v2/orgs/%s", orgID),
-			"self": fmt.Sprintf("/api/v2/orgs/%s/secrets", orgID),
-		},
-		Secrets: ks,
+		Organization: o,
 	}
 }
 
 // handlePostOrg is the HTTP handler for the POST /api/v2/orgs route.
 func (h *OrgHandler) handlePostOrg(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	req, err := decodePostOrgRequest(ctx, r)
-	if err != nil {
-		h.HandleHTTPError(ctx, err, w)
-		return
-	}
-
-	if err := h.OrganizationService.CreateOrganization(ctx, req.Org); err != nil {
-		h.HandleHTTPError(ctx, err, w)
+	var org influxdb.Organization
+	if err := h.API.DecodeJSON(r.Body, &org); err != nil {
+		h.API.Err(w, err)
 		return
 	}
-	h.log.Debug("Org created", zap.String("org", fmt.Sprint(req.Org)))
 
-	if err := encodeResponse(ctx, w, http.StatusCreated, newOrgResponse(req.Org)); err != nil {
-		logEncodingError(h.log, r, err)
+	if err := h.OrgSVC.CreateOrganization(r.Context(), &org); err != nil {
+		h.API.Err(w, err)
 		return
 	}
-}
+	h.log.Debug("Org created", zap.String("org", fmt.Sprint(org)))
 
-type postOrgRequest struct {
-	Org *influxdb.Organization
-}
+	h.API.Respond(w, http.StatusCreated, newOrgResponse(org))
 
-func decodePostOrgRequest(ctx context.Context, r *http.Request) (*postOrgRequest, error) {
```

Comment:
this PR axes a bunch of unneccsary indirection, reduce the jumping around and make it easier to follow within the handler itself.